### PR TITLE
ETH-443 - automatically grant permission to streams from project registry

### PR DESCRIPTION
**Pull request by samigabor**
ProjectRegistry has the trusted role in StreamRegistry and can use the trusted functions

### CODE REVIEW

File: packages/hub-contracts/contracts/ProjectRegistry/ProjectRegistry.sol [hunk 1]
```diff
@@ -335,7 +351,6 @@ contract ProjectRegistry is Initializable, UUPSUpgradeable, ERC2771ContextUpgrad
 
     function _addStream(bytes32 projectId, string calldata streamId) private {
         require(streamRegistry.hasPermission(streamId, _msgSender(), IStreamRegistry.PermissionType.Grant), "error_noGrantPermissionForStream");
-        _grantSubscribeForStream(streamId, address(this));
```
Comment (jtakalai) [inline packages/hub-contracts/contracts/ProjectRegistry/ProjectRegistry.sol:338]: Why was this needed before?

**Review by jtakalai — APPROVED** [review]

File: packages/hub-contracts/contracts/ProjectRegistry/ProjectRegistry.sol (hunk 1 above)
Comment (samigabor) [reply packages/hub-contracts/contracts/ProjectRegistry/ProjectRegistry.sol:338]: It would enable the Grant permission to project registry on the stream. This way when a user would buy subscription to a project, project registry would be able to grant permissions to the users buying the project. It was the mechanism replaced by the trusted role given to project registry on the stream registry contract.
